src/codegen.cpp

```diff
@@ -5971,6 +5979,36 @@ static std::unique_ptr<Module> emit_function(
             }
         }
     }
+
+    std::vector<MDNode*> aliasscopes;
+    MDNode* current_aliasscope = nullptr;
+    std::vector<Metadata*> scope_stack;
+    std::vector<MDNode*> scope_list_stack;
+    {
+        size_t nstmts = jl_array_len(src->code);
+        aliasscopes.resize(nstmts + 1, nullptr);
```

Comment:
This block and the for-loop the part that Yingbo and I were the most unsure about. We mimicked the `+ 1` from the linetable construct above.